lib/model/model.go

```diff
@@ -1338,6 +1339,14 @@ func (m *model) ccHandleFolders(folders []protocol.Folder, deviceCfg config.Devi
 	}
 
 	indexSenders.removeAllExcept(seenFolders)
+	if expired, err := m.db.ExpirePendingFolders(deviceID, handleTime); err != nil {
```

Comment:
I don't like this mechanism, as it's not obvious from this what gets added or removed.
I think you should explicitly, check what's in the CC, check what's in the database, get a list of what needs to be removed, remove it, potentially debug log it.

Reply:
> I don't like this mechanism, as it's not obvious from this what gets added or removed.

Strange, I like it very much for its efficiency and simplicity.  A one-line comment and maybe renaming the function to `ExpirePendingFoldersBeforeTime` should be enough to make it clearer?

> I think you should explicitly, check what's in the CC, check what's in the database, get a list of what needs to be removed, remove it, potentially debug log it.

That's surely more code and more runtime overhead, but I don't see what we actually gain from juggling with lists here?  IMO the current solution is dead simple and gets the job done, no need for anything more complex.

> Now it's some sort of magical "now" value that goes in, and magic happens underneath, with no clarify what effect that has.

Nothing magical: Note when the CC is received, update all contained folders' timestamps, delete pending folders which were not just touched.  Will add a comment to make that more obvious.